share/availability/discovery/discovery.go

```diff
@@ -348,6 +352,9 @@ func (d *Discovery) Advertise(ctx context.Context) {
 
 			select {
 			case <-timer.C:
+				if !timer.Stop() {
+					<-timer.C
+				}
```

Comment:
@walldiss, this is not needed before Reset in this case, as we just drained C

Additionally, this change introduces more flakes which I am in the process of debugging. Removing this exact locs helped.

Reply:
You are right. Timer is already stoped and channel drain at that point